util/buildRelease/smokeTest

```diff
@@ -118,6 +118,12 @@ num_procs=$($CHPL_HOME/util/buildRelease/chpl-make-cpu_count)
 # in $CHPL_HOME where the hello*.chpl tests are copied and run.
 export CHPL_CHECK_INSTALL_DIR=$CHPL_HOME
 
+# mason currently requires CHPL_COMM=none -- https://github.com/chapel-lang/chapel/issues/12626
+COMM=`$CHPL_HOME/util/chplenv/chpl_comm.py`
+if [ "$COMM" != "none" ]; then
+  CHPL_SMOKE_SKIP_MAKE_MASON=${CHPL_SMOKE_SKIP_MAKE_MASON:-true}
+fi
```

Comment:
Placing this before the first use of `CHPL_SMOKE_SKIP_MAKE_MASON` would prevent building re2 unnecessarily for comm!=none builds.